src/wasm/literal.cpp

```diff
@@ -37,6 +37,10 @@ Literal::Literal(Type type) : type(type) {
     assert(type != Type::unreachable && (!type.isRef() || type.isNullable()));
     if (type.isException()) {
       new (&exn) std::unique_ptr<ExceptionPackage>();
+    } else if (isGCData()) {
+      new (&gcData) std::shared_ptr<GCData>();
+    } else if (type.isRtt()) {
+      new (&rttSupers) auto(std::make_unique<RttSupers>());
```

Comment:
I don't think that's a valid change. The current code calls `make_unique` which mallocs some new data and points to it. The suggested change would not allocate anything, the `unique_ptr` is null.

About `auto`, good question... Removing `auto` leads to
```cpp
src/wasm/literal.cpp:80:22: error: expected type-specifier
   80 |     new (&rttSupers) std::make_unique<RttSupers>(*other.rttSupers);
      |                      ^~~
```
I guess `auto` in C++ makes that work somehow, but I have no idea how... :shrug:

Reply:
Oh, this is confusing because the desired behavior is different between the `ExceptionPayload` and the `RttSupers`. The reason `auto` is necessary here is because `std::make_unique` is not a constructor. If you had been able to change it to `std::unique_ptr`, you could have gotten rid of the `auto`.

Reply:
It seems the lines above which are for `GCData` and `ExceptionPackage` are fine even if they don't allocate anything? What's the difference?

Reply:
An `exn` can be null, I think that's the difference? `rttSupers` is always present. It would be just `RttSupers rttSupers;` inline, except that grows the size of all literals, so it's moved to be outside.

Reply:
Why is `rttSupers` always present? It is necessary only for rtts I think..? As `exn` is only necessary for exceptions. Maybe I'm missing something...

Reply:
Sorry, I meant - it's always present for an RTT. If the type is RTT, that is not null.

Reply:
I think it would be good to make the difference between this case and the exception package case explicit by adding a comment along the lines of `// Allocate an empty `rttSupers` vector`.

Reply:
It seems it is only me but I still don't understand why `exn` and `rttSupers` should be allocated in a different way... Both exist only in some of the cases, and why is it OK to leave one as a null pointer here in the default constructor and not OK for the other..?

Reply:
Does `exn` always exist if the type is of an exception? Then we can simplify it to be like `rttSupers`.

Reply:
It looks like that's not the case. `exn` can be `null`, and that indicates a null value in fact.

So `exn` and `rttSupers` are fundamentally different - there is never a `null` in `rttSupers`.

Reply:
Ah I think now I understand. Thanks for your patient explanation!